src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx

```diff
@@ -15,14 +17,28 @@ function ResultFiltersAndShare({
 }: ResultFiltersAndShareProps) {
   const { state } = useContext(QuizContext);
   const matchedOptions = state?.quiz?.matchedOptions || [];
+  const {
```

Comment:
I see the logic here is duplicated again in HooksTemplate. Can we abstract it in a function or a utility hook instead?

Reply:
What do you think about making summary it's own text component? 🤔

Reply:
Yes, using a Component does seem to be a more effective approach. I have completed the refactoring using the Component for the summary [4408a10](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/168/commits/4408a10b28051fee058364be910c0063b4d9d002)